src/integrations/Elastic.Apm.AspNetFullFramework/ElasticApmModule.cs

```diff
@@ -631,8 +631,8 @@ private void FillSampledTransactionContextUser(HttpContext context, ITransaction
 				return;
 
 			var user = new User { UserName = userIdentity.Name };
-
-			if (context.User is ClaimsPrincipal claimsPrincipal)
+			var sqlRoleProvider = System.Web.Security.Roles.Providers.Cast<object>().Any(provider => provider.GetType().Name == "SqlRoleProvider");
```

Comment:
There is an issue with `System.Web.Security.Roles` for a classic application developed with `ASPNET identity 2`. This will throw an exception - *The Role Manager feature has not been enabled* - since `<roleManager>` attribute in `web.config` is *NOT* used for an app with `ASPNET identity 2`

The correct way is to first check if `System.Web.Security.Roles.Enabled`. If it is NOT enabled, `sqlRoleProvider` variable should be false and stop immediately instead of trying to look into `System.Web.Security.Roles.Providers`. This will cause the exception - `The Role Manager feature has not been enabled` - to be thrown

Reply:
exception for your reference

```
[2024-08-13 16:23:15.175 -07:00][27][Error] - {ElasticApmModule.#2} Processing EndRequest event failed +-> Exception: System.Configuration.Provider.ProviderException: The Role Manager feature has not been enabled.
   at System.Web.Security.Roles.EnsureEnabled()
   at Elastic.Apm.AspNetFullFramework.ElasticApmModule.FillSampledTransactionContextUser(HttpContext context, ITransaction transaction) in /home/runner/work/apm-agent-dotnet/apm-agent-dotnet/src/integrations/Elastic.Apm.AspNetFullFramework/ElasticApmModule.cs:line 634
   at Elastic.Apm.AspNetFullFramework.ElasticApmModule.ProcessEndRequest(Object sender) in /home/runner/work/apm-agent-dotnet/apm-agent-dotnet/src/integrations/Elastic.Apm.AspNetFullFramework/ElasticApmModule.cs:line 608
   at Elastic.Apm.AspNetFullFramework.ElasticApmModule.OnEndRequest(Object sender, EventArgs e) in /home/runner/work/apm-agent-dotnet/apm-agent-dotnet/src/integrations/Elastic.Apm.AspNetFullFramework/ElasticApmModule.cs:line 332
```